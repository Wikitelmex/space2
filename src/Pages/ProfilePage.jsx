import React from 'react';
import MyMissionsComponent from '../Components/MyMissionsComponent';
import MyRockets from '../Components/MyRockets';

const ProfilePage = () => {
  const PageName = <h1>Profile Page</h1>;

  return (
    <div>
      {PageName}
      <div className="d-flex flex-row col-12">
        <div className="col-6 p-3">
          <h2>My Missions</h2>
          <MyMissionsComponent />
        </div>
        <div className="col-6 p-3">
          <h2>My Rockets</h2>
          <MyRockets />
        </div>
      </div>
    </div>
  );
};

export default ProfilePage;
