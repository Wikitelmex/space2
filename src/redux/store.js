import { createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

// import reducers here
import missionsReducer from './missions/missionReducer.js';

const mixReducer = combineReducers({
  // add reducers here
  missions: missionsReducer,
});

const store = createStore(
  mixReducer,
  applyMiddleware(logger, thunk),
);

export default store;
