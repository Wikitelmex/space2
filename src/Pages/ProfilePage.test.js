import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import ProfilePage from './ProfilePage';

import store from '../redux/store';

afterEach(cleanup);

it('renders profile page with redux', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <ProfilePage />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
