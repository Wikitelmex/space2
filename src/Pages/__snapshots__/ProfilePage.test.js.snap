// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mission page with redux 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Profile Page
    </h1>
    <div>
      <h2>
        My Missions
      </h2>
      <div>
        No Missions loaded yet...
      </div>
    </div>
    <div>
      <h2>
        My Rockets
      </h2>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders profile page with redux 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Profile Page
    </h1>
    <div>
      <h2>
        My Missions
      </h2>
      <div>
        No Missions loaded yet...
      </div>
    </div>
    <div>
      <h2>
        My Rockets
      </h2>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
